File: non_constructible_change/non_constructible_change.py
# O (n log(n)) time O(1) space
def non_constructible_change(coins):
  coins.sort()
  current_change_created = 0

  for coin in coins:
    if coin > current_change_created + 1:
      return current_change_created + 1
    current_change_created += coin

  return current_change_created + 1

print(non_constructible_change([5, 7, 1, 1, 2, 3, 22]))

File: non_constructible_change/non_constructible_change.ts
function nonConstructibleChange(coins: number[]): number {
    coins.sort((a, b) => a - b);
    let currentChangeCreated = 0;

    for (const coin of coins) {
      if (coin > currentChangeCreated + 1) {
        return currentChangeCreated + 1;
      }
      currentChangeCreated += coin;
  }

  return currentChangeCreated + 1;
}

const coins = [5, 7, 1, 1, 2, 3, 22];
console.log(nonConstructibleChange(coins));  
